internal/api/client/config.go

```diff
@@ -0,0 +1,127 @@
+package client
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+
+	"github.com/flightctl/flightctl/internal/crypto"
+	oscrypto "github.com/openshift/library-go/pkg/crypto"
+	"sigs.k8s.io/yaml"
+)
+
+// Config holds the information needed to connect to a FlightCtl API server
+type Config struct {
```

Comment:
The structure makes sense to me, we talked about trying to unify the cmdline tool config and agent config, we could do that eventually.

This config could be easily extended (AuthInfo section) to user/address/token (o whatever makes sense...) of a keycloak server for authentication when necessary,